lib/common/test/unit-tests/errors.ts

```diff
@@ -0,0 +1,189 @@
+			});
+
+			it("loggerLevel is TRACE", async () => {
+				const { action, printCommandHelpSuggestion } = setupTest({ err: new Error(errMsg) });
+				logger.loggerLevel = LoggerLevel.TRACE;
+				await errors.beginCommand(action, printCommandHelpSuggestion);
+				assertCallStack();
+				assertProcessExited();
+				assertPrintCommandHelpSuggestionIsNotCalled();
+			});
+
+			it("loggerLevel is DEBUG", async () => {
+				const { action, printCommandHelpSuggestion } = setupTest({ err: new Error(errMsg) });
+				logger.loggerLevel = LoggerLevel.DEBUG;
+				await errors.beginCommand(action, printCommandHelpSuggestion);
+				assertCallStack();
+				assertProcessExited();
+				assertPrintCommandHelpSuggestionIsNotCalled();
+			});
+		});
+
+		it("colorizes the error message to stderr when the terminal is interactive", async () => {
+			const { action, printCommandHelpSuggestion } = setupTest({ err: new Error(errMsg) });
+			isInteractive = true;
+			await errors.beginCommand(action, printCommandHelpSuggestion);
+			assert.equal(logger.errorOutput, `\x1B[31;1m${errMsg}\x1B[0m\n`);
+			assertProcessExited();
+			assertPrintCommandHelpSuggestionIsNotCalled();
+		});
+
+		it("prints message on stdout instead of stderr when printOnStdout is set in the Error", async () => {
+			const errObj: any = new Error(errMsg);
+			errObj.printOnStdout = true;
+			const { action, printCommandHelpSuggestion } = setupTest({ err: errObj });
+			await errors.beginCommand(action, printCommandHelpSuggestion);
+			assert.equal(logger.errorOutput, "");
+			assert.equal(logger.output, `${errMsg}\n`);
+			assertProcessExited();
+			assertPrintCommandHelpSuggestionIsNotCalled();
+		});
+
+		it("suggests how to show command help when error's suggestCommandHelp is set", async () => {
+			const errObj: any = new Error(errMsg);
+			errObj.suggestCommandHelp = true;
+			const { action, printCommandHelpSuggestion } = setupTest({ err: errObj });
+			await errors.beginCommand(action, printCommandHelpSuggestion);
+			assert.equal(logger.errorOutput, `${errMsg}\n`);
+			assertProcessExited();
+			assert.isTrue(isPrintCommandHelpSuggestionExecuted, "printCommandHelpSuggestion should be called when the action fails with an error object for which suggestCommandHelp is true.");
+		});
+
+		it("exits with passed exit code when the error has errorCode set to number", async () => {
+			const errObj: any = new Error(errMsg);
+			errObj.errorCode = 222;
+			const { action, printCommandHelpSuggestion } = setupTest({ err: errObj });
+			await errors.beginCommand(action, printCommandHelpSuggestion);
+			assert.equal(logger.errorOutput, `${errMsg}\n`);
+			assert.isTrue(isProcessExitCalled, "When the action fails, process.exit must be called.");
+			assert.equal(processExitCode, errObj.errorCode, `When the action fails, process.exit must be called with ${errObj.errorCode}.`);
+		});
+
+		it("exits with default exit code code when the error has errorCode set to number", async () => {
```

Comment:
the test name is wrong - it should be `exits with default exit code code when the error has errorCode set to string`